Build/QIR_validation.ps1

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = 'Stop'
+
+& "$PSScriptRoot/set-env.ps1"
+$all_ok = $True
+
+function Build-One {
+    param(
+        $project
+    );
+
+    $project_directory = $(Split-Path $project -Parent)
+    $project_name = $(Split-Path $project -LeafBase)
+
+    pushd $project_directory
```

Comment:
`pushd` may fail on some macOS and Linux systems, as they may have binaries with the same name that override the shell definition of `pushd`. In general, `Push-Location` and `Pop-Location` are more portable cross-platform.

Reply:
changed to Push-Location and Pop-Location